src/lib/sql/sql_pipeline.cpp

```diff
@@ -0,0 +1,136 @@
+#include <utility>
+
+#include "SQLParser.h"
+#include "concurrency/transaction_manager.hpp"
+#include "logical_query_plan/lqp_translator.hpp"
+#include "optimizer/optimizer.hpp"
+#include "sql_pipeline.hpp"
+#include "sql_translator.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+SQLPipeline::SQLPipeline(const std::string& sql, std::shared_ptr<TransactionContext> transaction_context) : _sql(sql) {
+  if (transaction_context != nullptr) {
+    _transaction_context = transaction_context;
+  } else {
+    _transaction_context = TransactionManager::get().new_transaction_context();
```

Comment:
This is somewhat inconsistent with what we do below where we can pass validate / don't validate. I believe that there are three cases:

(1) We don't want MVCC. Neither the caller (console/user) nor the pipeline creates a context. Validates are not added.
(2) The user passes a context. Validates are added. The pipeline is responsible for a rollback and informing the user of a rollback.
(3) The user passes no context, but wants MVCC. Validates are added. The pipeline is responsible for a rollback and informing the success / failure.

Reply:
So you suggest not passing validate but handling it internally?

Reply:
Exactly. The presence of a context (either user-provided or automatically generated) defines whether we validate or not.